src/main/resources/data/computercraft/lua/rom/programs/poster.lua

```diff
@@ -1,94 +1,180 @@
+  
+  term.setTextColor(colors.gray)
+  print("Options for .2dja: [ <index> | <from> <to> ]")
+  
+  term.setTextColor(colors.white)
+  print(("Type '%s stop' to cancel the print task"):format(programName))
 end
 
 -- Find a printer
 local printer = peripheral.find("poster_printer")
 if not printer then
-  error("No printer found", 0)
+  error("No printer found.", 0)
 end
 
 -- Validate arguments
-local args = { ... }
-if #args < 1 or #args > 2 then
+local tArgs = { ... }
+if #tArgs < 1 or #tArgs > 4
+or tArgs[1]:lower() == "help" then
   printUsage()
   return
 end
 
 -- Stop the printer by running `poster stop`
-if args[1] == "stop" then
+if tArgs[1] == "stop" then
   print("Stopping printer...")
   printer.stop()
   return
 end
 
-local count = 1
-if #args == 2 then
-  count = tonumber(args[2])
-  if not count or math.floor(count) < 1 then
+local status = printer.status()
+if status == "busy" then
+  error("Printer is busy.", 0)
+end
+
+-- Convert string to number and validate it.
+local function getNum(str)
+  local num = tonumber(str)
+  if not num or math.floor(num) < 1 then
+    return
+  end
+  return math.floor(num)
+end
+
+-- Get count.
+local nCount = 1
+if #tArgs > 1 then
+  nCount = getNum(tArgs[2])
+  if not nCount then
     printUsage()
     return
   end
-  count = math.floor(count)
 end
 
--- Load the 2dj file
-local filename = args[1]
+-- Get index, if given.
+local tIndex = { bAll = true, from = nil, to = nil }
+if #tArgs >= 3 then
+  local num1,num2 = getNum(tArgs[3]), getNum(tArgs[4])
+  if not num1 or not num2 then
+    printUsage()
+    return
+  end
+  
+  tIndex.from = math.min(num1,num2)
+  tIndex.to = math.max(num1,num2)
+  
+  -- Index, not interval
+  if not tIndex.to then
+    tIndex.to = tIndex.from
+  end
+  
+  if tIndex.from then
+    tIndex.bAll = false
+  end
+end
+
+local sFilename = tArgs[1]
+-- Loads from local file or Url
+local function loadPoster(sName)
+  local stream
 
-local function load2djFile(name)
-  local f
-  if http and name:match("^https?://") then
+  if http and sName:match("^https?://") then
     print("Downloading...")
-    f = http.get{ url = name, binary = true }
+    stream = http.get{ url = sName, binary = true }
   else
-    name = shell.resolve(name)
-    if not fs.exists(name) then
-      -- Try to find a .2dj file if the user didn't specify the extension
-      if not filename:match(".2dj$") then
-        local newName = name .. ".2dj"
-        if fs.exists(newName) then
-          return load2djFile(newName)
+    print("Loading...")
+    sName = shell.resolve(sName)
+    if not fs.exists(sName) then
+      -- Try to find a .2dj or .2dja file if the extension did not got specified 
+      if not sFilename:match(".2dj$") or not sFilename:match(".2dja$") then
+        for _,sEnd in ipairs({".2dj", ".2dja"}) do
+          local newName = sName..sEnd
+          if fs.exists(newName) then
+            return loadPoster(newName)
+          end
         end
       end
```

Comment:
If both exist, we should not assume one or the other, we should error and force the user to be more specific.